src/pages/SixCoinPuzzle.js

```diff
@@ -0,0 +1,131 @@
+    const leftCoin = selectedCoin - 1;
+    const jumpLeftCoin = selectedCoin - 2;
+    let temp = coinTypeArray;
+    // first check that coin is eligible to move left, then either move one space left
+    // or jump two spaces left
+    if (coinTypeArray[selectedCoin] === 1) {
+      toast.error("cannot move silver coin left", {
+        position: "bottom-center",
+      });
+    } else if (coinTypeArray[leftCoin] === 0) {
+      swapCoins(temp, selectedCoin, leftCoin);
+    } else if (coinTypeArray[jumpLeftCoin] === 0) {
+      swapCoins(temp, selectedCoin, jumpLeftCoin);
+    } else {
+      toast.error("no available space", { position: "bottom-center" });
+    }
+  };
+  const handleMoveRight = () => {
+    const rightCoin = selectedCoin + 1;
+    const jumpRightCoin = selectedCoin + 2;
+    let temp = coinTypeArray;
+    // first check that coin is eligible to move right, then either move one space right
+    // or jump two spaces right
+    if (coinTypeArray[selectedCoin] === 2) {
+      toast.error("cannot move gold coin right", {
+        position: "bottom-center",
+      });
+    } else if (coinTypeArray[rightCoin] === 0) {
+      swapCoins(temp, selectedCoin, rightCoin);
+    } else if (coinTypeArray[jumpRightCoin] === 0) {
+      swapCoins(temp, selectedCoin, jumpRightCoin);
+    } else {
+      toast.error("no available space", { position: "bottom-center" });
+    }
+  };
+  const checkCompletion = (array) => {
+    if (array.every((val, index) => val === winningArray[index])) {
+      toast.success("Congrats! You've completed the puzzle");
+    }
+  };
+  const resetPuzzle = () => {
+    setCoinTypeArray(startingArray);
+    setMoveCounter(0);
+    setCoinSelectText("Click to select a coin");
+  };
+
+  ArrowKeysReact.config({
+    left: () => {
+      handleMoveLeft();
+    },
+    right: () => {
+      handleMoveRight();
+    },
+    up: () => {
+      toast.error("cannot move up", { position: "bottom-center" });
+    },
+    down: () => {
+      toast.error("cannot move down", { position: "bottom-center" });
+    },
+  });
+
+  return (
+    <div className="coin-puzzle">
+      <div className="puzzle-name">Six Coin Puzzle</div>
+      <div className="objective-box">
+        Move all silver coins to the right side and all gold coins to the left
+      </div>
+      <div className="rules-box">
+        1. Silver coins must move to the right, gold coins must move to the left
+        <br></br>
+        2. Coins can jump one adjacent coin to land in an empty space
+      </div>
+      <div className="coin-select-text">
+        <p>{coinSelectText}</p>
+      </div>
+      <div className="coin-array" {...ArrowKeysReact.events} tabIndex="1">
+        {[...Array(numberOfCoinSlots).keys()].map((key) => (
+          <Coin
+            coinType={coinTypeArray[key]}
+            action={() => selectCoin(key)}
+            id={key}
+            isSelected={key === selectedCoin ? true : false}
+          />
+        ))}
```

Comment:
The second argument to `map()` is the array item index, so I think you could write this as:

```
[...Array(numberOfCoinSlots)].map((item, key) => {
<Coin
    coinType={coinTypeArray[key]}
    action={() => selectCoin(key)}
    id={key}
    isSelected={key === selectedCoin ? true : false}
/>
})
```

Also, any time you render a list of something in React you need to give it a `key` prop as well. So you would add `key={key}`